shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -206,17 +223,82 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
 
   /**
    * Generates the configuration docs.
+   * @param validate validate flag
    */
-  public static void generate() throws IOException {
+  public static void generate(boolean validate) throws IOException {
     Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
     defaultKeys.removeIf(key -> key.isHidden());
     String homeDir = new InstancedConfiguration(ConfigurationUtils.defaults())
         .get(PropertyKey.HOME);
     // generate CSV files
     String filePath = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
-    writeCSVFile(defaultKeys, filePath);
+    writeCSVFile(defaultKeys, filePath, validate);
+
+    if (validate) {
+      boolean hasDiff = false;
+      String[] fileNames = {"user-configuration.csv", "master-configuration.csv",
+          "worker-configuration.csv", "security-configuration.csv",
+          "common-configuration.csv", "cluster-management-configuration.csv"};
+
+      for (String fileName : fileNames) {
+        String tempPrefix = "temp-";
```

Comment:
this `temp-` prefix must match the same `temp-` prefix set in the `writeCSVFile` and `writeYMLFile`. if any one of these string prefixes changed, they must all be changed -> create a constant for this